src/pages/AddPurchase/AddPurchase.js

```diff
@@ -0,0 +1,133 @@
+import FoodList from '../../components/FoodList/FoodList'
+import SearchFood from '../../components/SearchFood/SearchFood'
+import styled from 'styled-components/macro'
+import { useState, useEffect } from 'react'
+import getFood from '../../services/getFood'
+import SumFootprint from '../../components/SumFootprint/SumFootprint'
+import PurchaseList from '../../components/PurchaseList/PurchaseList'
+import loadLocally from '../../lib/loadLocally'
+import saveLocally from '../../lib/savelocally'
+
+export default function AddPurchase() {
+  const [foodList, setFoodList] = useState([])
+  const [searchInput, setSearchInput] = useState('')
+  const [purchasedFood, setPurchasedFood] = useState(
+    loadLocally('purchasedFood') ?? []
+  )
+  const [foodListModal, setFoodListModal] = useState(false)
+  const [carbonFootprintSum, setCarbonFootprintSum] = useState(
+    loadLocally('carbonFootprintSum') ?? 0
+  )
+  const [pointerPosition, setPointerPosition] = useState(0)
+
+  useEffect(() => {
+    getFood()
+      .then((data) => setFoodList(data))
+      .catch((error) => console.log(error))
+  }, [])
+
+  function handleChange(event) {
+    setSearchInput(event.target.value)
+  }
+
+  function toggleFoodListModal() {
+    setFoodListModal(!foodListModal)
+  }
+
+  const filteredFoodList = foodList.filter((foodListItem) =>
+    foodListItem.food.toLowerCase().includes(searchInput.toLowerCase())
+  )
+
+  function handleAddFood(id) {
```

Comment:
This function name could also be improved, as it seems like more is going on within the function than only adding food (it also updates the total footprint sum). Maybe something like updateFoodListAndFootprintSum? :) Sometimes it is okay to have longer names